src/manifolds/Grassmann.jl

```diff
@@ -278,12 +278,12 @@ In this formula the $\operatorname{atan}$ is meant elementwise.
 """
 log(::Grassmann, ::Any...)
 
-function log!(M::Grassmann, X, p, q)
+function log!(::Grassmann{n,k}, X, p, q) where {n,k}
     z = q' * p
     At = q' - z * p'
     Bt = z \ At
     d = svd(Bt')
-    return copyto!(X, d.U * Diagonal(atan.(d.S)) * d.Vt)
+    return copyto!(X, d.U[:, 1:k] * Diagonal(atan.(d.S[1:k])) * d.Vt[1:k, :])
```

Comment:
Perhaps something like this to reduce number of allocations?
```suggestion
    return X .= view(d.U, :, 1:k) * Diagonal(atan.(view(d.S, 1:k))) * view(d.Vt, 1:k, :))
```